solr/core/src/java/org/apache/solr/schema/LatLonPointSpatialField.java

```diff
@@ -75,8 +77,16 @@ protected SpatialStrategy newSpatialStrategy(String fieldName) {
     return new LatLonPointSpatialStrategy(ctx, fieldName, schemaField.indexed(), schemaField.hasDocValues());
   }
   
-  public String geoValueToStringValue(long value) {
-    return new String(decodeLatitudeCeil(value) + "," + decodeLongitudeCeil(value));
+  /**
+   * Converts to "lat, lon"
+   * @param value Non-null; stored location field data
+   * @return Non-null; "lat, lon" with 6 decimal point precision
+   */
+  public static String decodeDocValueToString(long value) {
+    double latitudeDecoded = BigDecimal.valueOf(decodeLatitude((int) (value >> 32))).setScale(6, HALF_UP).doubleValue();
```

Comment:
Lets have some comments explaining why this algorithm here is what it is.  Why HALF_UP?

Can we skip the doubleValue and just do toPlainString (avoiding exponent notation of toString) since we're composing a string in the end?  In other words, lets avoid the pointless double primitive intermediary.

Reply:
HALF_UP is only for ceil, I will remove the rounding.

Reply:
I'm not arguing with your choice; just trying to understand.